includes/Core/Modules/REST_Modules_Controller.php

```diff
@@ -435,9 +436,28 @@ private function get_rest_routes() {
 							if ( ! $module instanceof Module_With_Settings ) {
 								return new WP_Error( 'invalid_module_slug', __( 'Module does not support settings.', 'google-site-kit' ), array( 'status' => 400 ) );
 							}
-							return new WP_REST_Response( $module->get_settings()->get() );
+
+							if ( $can_manage_options() ) {
+								return new WP_REST_Response( $module->get_settings()->get() );
+							}
+
+							$settings = $module->get_settings();
+							if ( $settings instanceof Setting_With_ViewOnly_Keys_Interface ) {
+								$view_only_keys = $settings->get_view_only_keys();
+								$settings       = $settings->get();
+
+								if ( empty( $view_only_keys ) || empty( $settings ) ) {
```

Comment:
This condition is unnecessary because `array_intersect_key` below will return an empty array in either case anyways, and `array_flip` is also safe to use with an empty array 👍 `$settings` will always be an array and never be empty since the full value will always be an array with one or more key-value pairs.